primefaces/src/main/java/org/primefaces/component/datatable/DataTable.java

```diff
@@ -354,9 +354,10 @@ else if ("rowUnselect".equals(eventName) || "rowUnselectCheckbox".equals(eventNa
             else if ("page".equals(eventName) || "virtualScroll".equals(eventName) || "liveScroll".equals(eventName)) {
                 int rows = getRowsToRender();
                 int first = Integer.parseInt(params.get(clientId + "_first"));
+                int rowsPerPage = Integer.parseInt(Objects.toString(params.get(clientId + "_rows"), "-1"));
```

Comment:
should we make it nullable instead of -1?

Reply:
The only reason i left it lower `int` with a value was to be consistent with the other fields...and since for `virtuslScroll` and `LiveScroll` its a PageEvent with no rows i figured just make it -1 and leave it an `int` vs making it an `Integer`?

```java
                int rows = getRowsToRender();
                int first = Integer.parseInt(params.get(clientId + "_first"));
                int rowsPerPage = Integer.parseInt(params.get(clientId + "_rows"));
                int page = rows > 0 ? (int) (first / rows) : 0;

                PageEvent pageEvent = new PageEvent(this, behaviorEvent.getBehavior(), page, rowsPerPage);
```

Reply:
i see
i would prefer nullable, its more clear

Reply:
actually we should rethink other fields, too then

Reply:
OK I am going to refactor this to be `Integer` and allow NULL.